provision-cluster/delete.js

```diff
@@ -1,14 +1,37 @@
-const core = require('@actions/core');
-const github = require('@actions/github');
+'use strict';
 
-try {
+const core = require('@actions/core')
+const github = require('@actions/github')
+
+const registry = require('./registry.js')
+
+async function do_delete() {
   // inputs are defined in action metadata file
-  const distribution = core.getInput('distribution');
-  const version = core.getInput('version');
-  console.log(`Deleting ${distribution} ${version}!`);
-  // Get the JSON webhook payload for the event that triggered the workflow
-  //const payload = JSON.stringify(github.context.payload, undefined, 2)
-  //console.log(`The event payload: ${payload}`);
-} catch (error) {
-  core.setFailed(error.message);
+  const distribution = core.getInput('distribution')
+  let clusterName = core.getState(registry.CLUSTER_NAME)
+
+  let provider = registry.getProvider(distribution)
+
+  let promises = []
+  promises.push(expire(provider))
+
+  if (typeof clusterName !== typeof undefined && clusterName !== "") {
+    core.notice(`Deleting ${distribution} cluster ${clusterName}!`)
+    promises.push(delete_allocated(provider, clusterName))
+  }
+
+  return Promise.all(promises)
```

Comment:
Is it more appropriate to use `Promise.allSettled()`? 

`Promise.all()` will terminate if one promise fails, even if other promises are still running.

Reply:
I had it as allSettled before, but that ended up masking a bug in the code. I switched it to all so that any bugs would surface and we would find out about them.

Reply:
I did add a unit test for the particular bug that was surfaced, but the reason we want to fail fast is not just bugs, errors of any kind could get masked, kinda like having a `try { ... } catch (Throwable t) { // do nothing }` in java.

Say for example someone changes the permissions of the service account we're using so that it can't delete clusters. We want the resulting error to be propagated so that we know we're leaking clusters ASAP instead of having to wait for the huge bill to tell us.